src/coreclr/jit/gentree.h

```diff
@@ -3279,6 +3283,149 @@ struct GenTreeStrCon : public GenTree
 #endif
 };
 
+/* gtVecCon -- vector  constant (GT_CNS_VEC) */
+
+struct GenTreeVecCon : public GenTree
+{
+    union {
+        simd8_t  gtSimd8Val;
+        simd16_t gtSimd16Val;
+        simd32_t gtSimd32Val;
+    };
+
+    var_types GetSimdType() const
+    {
+        var_types simdType = gtType;
+
+#if defined(FEATURE_SIMD)
+        if (simdType == TYP_SIMD12)
+        {
+            simdType = TYP_SIMD16;
+        }
+#endif // FEATURE_SIMD
+
+        return gtType;
+    }
+
+#if defined(FEATURE_HW_INTRINSICS)
+    static bool IsHWIntrinsicCreateConstant(GenTreeHWIntrinsic* node, simd32_t& simd32Val);
+
+    static bool HandleArgForHWIntrinsicCreate(GenTree* arg, int argIdx, simd32_t& simd32Val, var_types baseType);
+#endif // FEATURE_HW_INTRINSICS
+
+    bool IsAllBitsSet() const
+    {
+        switch (gtType)
+        {
+#if defined(FEATURE_SIMD)
+            case TYP_LONG: // TYP_SIMD8 may get retyped to TYP_LONG after lowering
+            case TYP_SIMD8:
+            {
+                return (gtSimd8Val.u64[0] == 0xFFFFFFFFFFFFFFFF);
+            }
+
+            case TYP_SIMD12:
+            case TYP_SIMD16:
+            {
+                return (gtSimd16Val.u64[0] == 0xFFFFFFFFFFFFFFFF) && (gtSimd16Val.u64[1] == 0xFFFFFFFFFFFFFFFF);
+            }
+
+            case TYP_SIMD32:
+            {
+                return (gtSimd32Val.u64[0] == 0xFFFFFFFFFFFFFFFF) && (gtSimd32Val.u64[1] == 0xFFFFFFFFFFFFFFFF) &&
+                       (gtSimd32Val.u64[2] == 0xFFFFFFFFFFFFFFFF) && (gtSimd32Val.u64[3] == 0xFFFFFFFFFFFFFFFF);
+            }
+#endif // FEATURE_SIMD
+
+            default:
+            {
+                unreached();
+            }
+        }
+    }
+
+    bool IsEqual(const GenTreeVecCon* other) const
```

Comment:
Nit: the convention on `GenTree` nodes is to have a `GenTree::Equals(GenTree*, GenTree*)` `static` method (e. g. `GenTreePhi::Equals`, `GenTreeFieldList::Equals`, `GenTreeCall::Equals`, etc).

Reply:
Fixed.